### PR TITLE
AD-13664: Handle errors and logs and write in a separate Logs file

### PR DESCRIPTION
**Pull request by Rakshitha-Sheshagiri**

### CODE REVIEW

File: src/utils/csv.ts [hunk 1]
```diff
@@ -1,8 +1,18 @@
+import { createObjectCsvWriter } from 'csv-writer';
 import csvToJson from 'csvtojson';
+import path from 'path';
```
Comment (vinothkannanmani) [inline src/utils/csv.ts:4]: why need a separate file for this. why not use utlis.ts

File: src/utils/csv.ts [hunk 2]
```diff
@@ -1,8 +1,18 @@
+import { createObjectCsvWriter } from 'csv-writer';
```
Comment (vinothkannanmani) [inline src/utils/csv.ts:1]: we can directly write a csv file without this npm

Comment (vinothkannanmani) [reply src/utils/csv.ts:1]: if you want this one please explain here.

File: src/utils/csv.ts (hunk 1 above)
Comment (Rakshitha-Sheshagiri) [reply src/utils/csv.ts:4]: csv.ts under utils folder already existed. So, thought of having the writeToCsvFile method here.

File: src/utils/csv.ts (hunk 2 above)
Comment (Rakshitha-Sheshagiri) [reply src/utils/csv.ts:1]: yes thanks, we can write to csv without the npm

File: src/commands/emailmigration/execute.ts [hunk 3]
```diff
@@ -120,6 +123,15 @@ export const handler: Handler = async (argv) => {
 
   const failedEmailMigrationWithReasons = [];
 
+  // write errors and logs to a csv file
+  const writeFailedEmailMigrationsToCsv = createWriteStream(
+    path.resolve(__dirname, `../../../logs/email_migration_${Date.now()}`),
```
Comment (vinothkannanmani) [inline src/commands/emailmigration/execute.ts:128]: I think the file extension is missing if possible please attach the CSV file

**Review by vinothkannanmani — CHANGES_REQUESTED** [review]

File: src/commands/emailmigration/execute.ts (hunk 3 above)
Comment (Rakshitha-Sheshagiri) [reply src/commands/emailmigration/execute.ts:128]: yes correct, thanks for noticing. I will include the file extension.

Comment (Rakshitha-Sheshagiri) [reply src/commands/emailmigration/execute.ts]: [email_migration_1666774240812.csv](https://github.com/Bluescape/bluescape-cli/files/9868478/email_migration_1666774240812.csv)

**Review by vinothkannanmani — APPROVED** [review]
LGTM

**Comment by Rakshitha-Sheshagiri** [comment]
https://jira.common.bluescape.com/browse/AD-13664

File: src/commands/emailmigration/execute.ts [hunk 4]
```diff
@@ -121,6 +123,16 @@ export const handler: Handler = async (argv) => {
 
   let failedEmailMigrationWithReasons = 0;
 
+  // write errors and logs to a csv file
+  mkdirSync(path.join(__dirname, '../../../logs'));
+  const writeFailedEmailMigrationsToCsv = createWriteStream(
+    path.resolve(__dirname, `../../../logs/email_migration_${Date.now()}.csv`),
```
Comment (prem15588) [inline src/commands/emailmigration/execute.ts:131]: Please add the file name we get from input + current timestamp.cs

**Review by prem15588 — APPROVED** [review]

**Pull request merged**